power_grid_model_c/power_grid_model_c/include/power_grid_model_c/dataset.h

```diff
@@ -83,6 +83,47 @@ PGM_API PGM_Idx PGM_dataset_info_elements_per_scenario(PGM_Handle* handle, PGM_D
  */
 PGM_API PGM_Idx PGM_dataset_info_total_elements(PGM_Handle* handle, PGM_DatasetInfo const* info, PGM_Idx component_idx);
 
+/**
+ * @brief Return if a component has attribute indications.
+ *
+ * Attribute indications are used to indicate the presence of meaningful attributes for a certain component in the
+ * dataset. It returns 1 for a component if all the elements of all the scenarios for this component are list-like in
+ * the serialized data.
+ *
+ * @param handle
+ * @param info A pointer to the info object.
+ * @param component_idx The index number of the component.
+ * @return 1 if the component has attribute indications, 0 if it does not.
+ */
+PGM_API PGM_Idx PGM_dataset_info_has_attribute_indications(PGM_Handle* handle, PGM_DatasetInfo const* info,
```

Comment:
here but also elsewhere: `has_attribute_indications` feels ambiguous, because it suggests that there are indications in the header, but it does not necessarily suggest that there are no in-body attributes.

Maybe it's good to rename this to something like `*has_only_attribute_indications` or `*has_packed_data` or `*is_packed` or `*only_preamble_attribute_indications` or something similar?

Reply:
The name makes sense from user perspective. `has_attribute_indications` means the user can rely on the `attribute_indications` to create buffers and not missing any meaningful data.

If `has_attribute_indications == false`. The user does not have such indications so that the user has to create buffers for all attributes or use row-based buffer.

Reply:
My issue is the following edge case:

```json
{
    "version": "1.0",
    "type": "input",
    "is_batch": false,
    "attributes": {
        "node": [
            "id",
            "u_rated"
        ]
    },
    "data": {
        "node": [
            [
                5,
                10500
            ],
            {
                "id": 6,
                "u_rated": 10500
            }
        ]
    }
}
```

As a user, i now expect `has_attribute_indications` to be `true` (because `node` is present in the header), but the logic says it's `false` (because there are maps in the node input).

Reply:
The point is, we should not focus on a particular serialized format. The idea of `has_attribute_indications` is from the information perspective, regardless of the actual serialization format and data.

It answers the following questions for the user: if I only create attribute buffers as in `attribute_indications`, am I guaranteed to retain all the relevant information from the dataset?

That's the reason I give this name. Currently, this condition just happens to be satisfied by "no-map in the data section in the msgpack/json format". But it does not stay that way. We can, for example, in the future to scan the actual map-like elements and still give the `has_attribute_indications` with `attribute_indications` of all the meaningful attributes.

Reply:
resolved cfr. discussion